### PR TITLE
Fix parser unit tests

### PR DESCRIPTION
**Pull request by antaljanosbenjamin**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

This pr actually does two things, but as they are small changes I thought it is better not to open two PRs:
1. Fixes how parser unit tests are printed to adhere the format that the result parser expects.
2. Update the PRQLParser unit tests to use the correct syntax from the update prql lib.

### How the printing of failed parser unit tests changed
Before:
```
[  FAILED  ] ParserPRQL/ParserTest.parseQuery/1, where GetParam() = ((ptr = 0x7030000a18f8, value = 8-byte object <48-D6 90-14 C0-55 00-00>), ParserTestCase input: from matches
filter start_date > @2023-05-30                 # Some comment here
derive [
  some_derived_value_1 = a + (b ?? 0),          # And there
  some_derived_value_2 = c + some_derived_value
]
filter some_derived_value_2 > 0
group [country, city] (
  aggregate [
    average some_derived_value_2,
    aggr = max some_derived_value_2,
  ]
)
derive place = f"{city} in {country}"
derive country_code = s"LEFT(country, 2)"
sort [aggr, -country]
take 1..20) (254 ms)
```
After:
```
[  FAILED  ] ParserPRQL/ParserTest.parseQuery/1, where GetParam() = ((ptr = 0x7f586399b5c8, value = 8-byte object <28-A0 F7-00 00-00 00-00>), ParserTestCase input: from matches\nfil ter start_date > @2023-05-30                 # Some comment here\nderive {\n  some_derived_value_1 = a + (b ?? 0),          # And there\n  some_derived_value_2 = c + some_derived_value\n}\nfilter some_derived_value_2 > 0\ngroup {country, city} (\n  aggregate {\n    average some_derived_value_2,\n    aggr = max some_derived_value_2\n  }\n)\nderive place = f"{city} in {country}"\nderive country_code = s"LEFT(country, 2)"\nsort {aggr, -country}\ntake 1..20) (7 ms)
```
Yes, it is less readable, but at least parsed correctly. E.g.:[#52792](https://github.com/ClickHouse/ClickHouse/pull/52792). Open its [unit tests](https://s3.amazonaws.com/clickhouse-test-reports/52792/be3b23f48cdefec8eea7842f2662c18a3820eb50/unit_tests__msan_.html), then the [run.log](https://s3.amazonaws.com/clickhouse-test-reports/52792/be3b23f48cdefec8eea7842f2662c18a3820eb50/unit_tests__msan_/run.log). Search for `[  FAILED  ]` and find the unnoticed failed PRQL tests.

### CODE REVIEW

File: src/Parsers/tests/gtest_Parser.cpp [hunk 1]
```diff
@@ -39,7 +39,9 @@ std::ostream & operator<<(std::ostream & ostr, const std::shared_ptr<IParser> pa
 
 std::ostream & operator<<(std::ostream & ostr, const ParserTestCase & test_case)
 {
-    return ostr << "ParserTestCase input: " << test_case.input_text;
+    std::string input_text{test_case.input_text};
```
Comment (rschu1ze) [inline src/Parsers/tests/gtest_Parser.cpp:42]: Maybe add a brief comment why newlines are removed? Just that future readers won't wonder.

**Review by rschu1ze — APPROVED** [review]

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 358f22d18a57ef85481742c45cb67c6c4515ac87 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/54670/358f22d18a57ef85481742c45cb67c6c4515ac87/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>✅ success</td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>⏳ pending</td></tr>
<tbody>
</table>

File: src/Parsers/tests/gtest_Parser.cpp (hunk 1 above)
Comment (antaljanosbenjamin) [reply src/Parsers/tests/gtest_Parser.cpp:45]: Done.

**Comment by antaljanosbenjamin** [comment]
I messed up the formatting and one of the tests... I will force-push the branch in order to get rid of unnecessary changes.

**Review by alexey-milovidov — APPROVED** [review]
Ok. This escaping is slightly incorrect because it adds ambiguity. But good for test outputs.

**Comment by antaljanosbenjamin** [comment]
> Ok. This escaping is slightly incorrect because it adds ambiguity. But good for test outputs.

Yes, I tried to minimize the ambiguity by using `\n`, so with an easy search and replace we can get back the original queries and we can utilize our current test output parser.

**Pull request merged**